src/components/menuBar/menuBarButtons.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import styled from "@emotion/styled";
+
+const ButtonContainer = styled.button`
+  line-height: 17px;
+  letter-spacing: -0.02em;
+  padding: 6px 12px;
+  border-radius: 0.5rem;
+`;
+
+interface MenuBarButtonsProps {
+  setSelectedButton: (name: string) => () => void;
+  selectedButton: string;
+}
+
+export const MenuBarButtons: React.FunctionComponent = (props: MenuBarButtonsProps) => {
+  const onSelectFilter = (name: string) => () => {
+    props.setSelectedButton(name);
+  };
+
+  const renderButton = (buttonName: string): JSX.Element => {
```

Comment:
The typing for this function is wrong but i am not sure how to fix it.

Type '{ setSelectedButton: Dispatch<SetStateAction<string>>; selectedButton: string; }' is not assignable to type 'IntrinsicAttributes & { children?: ReactNode; }'.
  Property 'setSelectedButton' does not exist on type 'IntrinsicAttributes & { children?: ReactNode; }'.

Reply:
This can be a element outside by itself, not necessary for it to be inside MenuBarButton. Same type as MenuBarButtons.

Reply:
you mean i should remove menuBarButtons.tsx?

Reply:
No, that is fine, the current way you are doing is more correct.